### PR TITLE
fix: Do not recover background streaming jobs whose table fragments have been marked as created

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fix assertion failure in background ddl test that was found in main-cron job: https://buildkite.com/risingwavelabs/main-cron/builds/1436#018cd00e-4ea4-4b6d-949e-d73a5a4fd08d:

> thread '<unnamed>' panicked at src/meta/src/manager/catalog/fragment.rs:588:9:
> assertion `left == right` failed
>   left: Created
>  right: Creating

The root cause is that the meta got killed after the streaming job was marked as `CREATED` but didn't have time to mark the status of catalog. In this case, we don't have to recover its tracking status but to finish the catalog directly.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by yezizp2012** [comment]
The background recovery test pasted.

File: ci/scripts/cron-e2e-test.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 set -euo pipefail
 
 source ci/scripts/common.sh
-export RUN_COMPACTION=1;
```
Comment (kwannoel) [inline ci/scripts/cron-e2e-test.sh:7]: Why disable this?

**Review by kwannoel — APPROVED** [review]
Rest LGTM, thanks!

**Review by BugenZhao — APPROVED** [review]

File: ci/scripts/cron-e2e-test.sh (hunk 1 above)
Comment (yezizp2012) [reply ci/scripts/cron-e2e-test.sh:7]: I found that a bug was introduced in #14259 when I ran the main-cron job. After discussion with @zwang28 , we decided to disable it temporarily and Zheng will open another PR to fix it.

**Pull request merged**